storage/innobase/log/log0recv.cc

```diff
@@ -1852,7 +1852,28 @@ recv_parse_or_apply_log_rec_body(
 	        Otherwise, redo will not find the key to decrypt
 		the data pages. */
 		if (page_no == 0 && !is_system_tablespace(space_id)) {
-			if (fil_write_encryption_parse(ptr, end_ptr,
+			byte* ptr_copy = ptr;
+			//increase by offset
+                        ptr_copy += 2;
+                        ulint len = mach_read_from_2(ptr_copy);
+                        ptr_copy += 2;
+                        if (end_ptr < ptr_copy + len)
+                                return NULL;
+
+			/* to only support encryption ='n' with PS */
+			if (memcmp(ptr_copy, ENCRYPTION_KEY_MAGIC_PS_V1,
+                                   ENCRYPTION_MAGIC_SIZE) == 0) {
+				// increase by  magic size, iv_len, space_id, offset,
+				ptr_copy += 3 + 2  + 4 + 2 ;
+				uint type = mach_read_from_1(ptr_copy);
+				if(type != CRYPT_SCHEME_UNENCRYPTED) {
+					ib::error() << "Can't take backup of "
+						"tablespace encrypted with "
+						"KEYRING encrypion";
```

Comment:
better to add  "<<" at start of every line (some compilers don't like this concatenation)